doc/releases/changelog-dev.md

```diff
@@ -77,9 +77,10 @@
 
 <h3>Improvements 🛠</h3>
 
-* The `JacobianProductCalculator` abstract base class and implementation `TransformJacobianProducts`
-  have been added to `pennylane.interfaces.jacobian_products`.
+* The `JacobianProductCalculator` abstract base class and implementations `TransformJacobianProducts`
+  and `DeviceDerivatives` have been added to `pennylane.interfaces.jacobian_products`.
```

Comment:
I see that the Calculator introduced in #4637 carries the corresponding name. 
I think we should discuss the naming here, because it is currently based on the underlying computational strategy rather than the computed quantities, which may lead to confusion imho.
Alternatively, we could be explicit in the differences between the two added classes.

Reply:
As you pointed out, we will end up with two classes:

1. Uses device provided jacobians and then performs the product (this PR)
2. Uses device provided jvp and vjps. (next pr)

I'd be happy to get suggestions for the names.  Maybe `DeviceDerivatives` for this one? Then its more of a break from the "jacobian product" name?